CMakeLists.txt

```diff
@@ -2014,6 +2024,11 @@ if(BUILD_TESTS)
                                   PROPERTIES COMPILE_FLAGS "-m32 ${RR_TEST_FLAGS}")
     endforeach(test)
 
+    if (x86ish)
+      configure_file("${CMAKE_CURRENT_SOURCE_DIR}/src/test/x86/gdb_avx512.c" "${CMAKE_CURRENT_BINARY_DIR}/32/gdb_avx512.c" COPYONLY)
+      set_source_files_properties("${CMAKE_CURRENT_BINARY_DIR}/32/${test}.c" PROPERTIES COMPILE_FLAGS "-m32 ${RR_TEST_FLAGS} -g3 -mavx512f")
```

Comment:
Do you really need `-mavx512f`? What happens if you remove that (and make this a regular test), and remove the `defined(__AVX512F__)` code in the test? I think it will probably just work? That test only uses AVX512 via inline assembly which I don't think requires `-mavx512f`.

Reply:
> Do you really need `-mavx512f`? What happens if you remove that (and make this a regular test), and remove the `defined(__AVX512F__)` code in the test? I think it will probably just work? That test only uses AVX512 via inline assembly which I don't think requires `-mavx512f`.

There's an interesting error that occurs when I remove the `-mavx512f` flag (and do not use `-march=native`): 

```
../src/test/x86/gdb_avx512.c:42:5: error: the register ‘zmm30’ cannot be clobbered in ‘asm’ for the current target
   42 |     asm volatile (
```

However, if I change the clobbered register to be something like `zmm9` and `zmm10` it _does_ compile without the flags, which means I can still test the contents of the registers is ok, but not all the way up to zmm31

Reply:
It worked just fine to remove it, had to change what registers get clobbered though, but that's probably just me being unfamiliar with inline assembly, because I have no idea why that had to happen.